flink-libraries/flink-table/src/test/scala/org/apache/flink/table/functions/aggfunctions/AggFunctionTestBase.scala

```diff
@@ -36,14 +36,23 @@ abstract class AggFunctionTestBase[T] {
 
   def aggregator: AggregateFunction[T]
 
+  def ifSupportRetraction: Boolean = true
+
   @Test
-  // test aggregate functions without partial merge
-  def testAggregateWithoutMerge(): Unit = {
+  // test aggregate and retract functions without partial merge
+  def testAccumulateAndRetractWithoutMerge(): Unit = {
     // iterate over input sets
     for ((vals, expected) <- inputValueSets.zip(expectedResults)) {
-      val accumulator = aggregateVals(vals)
-      val result = aggregator.getValue(accumulator)
+      val accumulator = accumulateVals(vals)
+      var result = aggregator.getValue(accumulator)
       validateResult(expected, result)
+
+      if (ifSupportRetraction) {
+        retractVals(accumulator, vals)
```

Comment:
We also need to check the retraction of a single value not all values (the `SumAggregator` retraction of all values was correct because the count was `0`).
For that we can accumulate all values and retract one value and compare the result against the result of accumulating all values except the retracted one.

Also check that retracting `null` does not have an effect.